framework/src/MonteCarloPlayerMT.cpp

```diff
@@ -0,0 +1,101 @@
+#include "MonteCarloPlayerMT.h"
+#include "GameTypes.h"
+#include "RandomPlayer.h"
+
+namespace Player {
+
+MonteCarloPlayerMT::MonteCarloPlayerMT() {
+    for(unsigned int i = 0; i < NUM_THREADS; ++i) {
```

Comment:
We don't have to add this now, but it may be beneficial in the future to use std::thread::hardware_concurrency() to check how many threads are available. Even if we don't use the maximum amount of threads, we want to be sure that we don't launch more threads than is available.

Reply:
I actually didn't realize that was a thing. added in next commit alongside the ability to set a max number of threads just in case you want to use less than what is available.